src/NServiceBus.Core/Encryption/RijndaelEncryptionService.cs

```diff
@@ -227,6 +227,11 @@ protected virtual bool TryGetKeyIdentifierHeader(out string keyIdentifier, IInco
             return context.Headers.TryGetValue(Headers.RijndaelKeyIdentifier, out keyIdentifier);
         }
 
+        protected virtual void GenerateIV(RijndaelManaged rijndael)
```

Comment:
But in theory nothing would prevent the implementor to also override mode, block size etc. on the passed in Rijndael instance which is kinda fishy for me. Following your argument of extensibility I see two approaches:

The rijndael managed is passed in, the method is renamed to configure rijndael and the methods sets everything on the instance and the calling method does nothing but just use it

or the method keeps its name and the calling method passes in a values required to be able to determine the right IV and the IV is returned by the method but all rijndael managed access remains in the calling method